### ISSUE TITLE
Book8088 support

### ISSUE THREAD

**Issue by PerditionC** [640-KB/GLaBIOS#44]
Current Book8088 build boots.  However, there is an error 6000 (keyboard controller and keyboard stuck error), pressing any key allows the computer to continue to boot and the keyboard seems to work as expected.  

Book8088 version 2 with VGA and V20 processor with only minimal testing.
Additionally, it has 2MB EMS ISA card and RTC ISA on port 70h.  When I get time, I want to try adding GLaTICK to the BIOS.

**Comment by 640-KB** [comment]
Thanks for giving it a try! I don't actually own one and haven't actually had many testers yet (I think just one early on). I'm tempted now on the book 8088 to simply skip those tests (for now at least) since we can be fairly confident the KBC should be working.

And yeah the "AT" build of GLaTICK should work just fine with a port 70h DS12x85/CMOS style RTC. Curious, is that an add-on card you are using? 

I greatly appreciate all of the testing. If you come across any other issues or something that isn't implemented please let me know.

640KB

**Comment by PerditionC** [comment]
Yes, ISA card from https://www.tindie.com/products/spark2k06/rtc-isa-8-bits-very-low-profile-2/ mapped to port 70h.  I stacked a few 28pin dip sockets and added a zif socket in my book8088 so I can easily swap out the BIOS chip and have two chips I can switch between for testing.  I happy to test any builds you want.  I got this so I can test the FreeDOS kernel more easily.

**Comment by 640-KB** [comment]
> I'm tempted now on the book 8088 to simply skip those tests (for now at least) since we can be fairly confident the KBC should be working.

Well, it appears that was exactly what I had previously done, and disabled those tests for the Book 8088 build, but perhaps re-enabled them to someday do more testing (this was a few months ago... so I don't remember):
https://github.com/640-KB/GLaBIOS/blob/24dd103ce354ddd2e6f1a913d420f72853366fdf/src/GLABIOS.ASM#L477

So, if you want to just uncomment that line and rebuilt it, that should "solve" the problem. :)  I'll do the same in the source tree as well.

> Yes, ISA card from https://www.tindie.com/products/spark2k06/rtc-isa-8-bits-very-low-profile-2/ mapped to port 70h. I stacked a few 28pin dip sockets and added a zif socket in my book8088 so I can easily swap out the BIOS chip and have two chips I can switch between for testing. I am happy to test any builds you want. I got this so I can test the FreeDOS kernel more easily.

Perfect!  I have that exact one and used it to test against, so know they work great together.  Really appreciate your help in testing too -- I'll take you up on that for sure.  Just let me know if you see anything that's not working right or anything else we might need to add for the Book 8088.

Thanks!

640KB

**Comment by 640-KB** [comment]
> Yes, ISA card from https://www.tindie.com/products/spark2k06/rtc-isa-8-bits-very-low-profile-2/ mapped to port 70h. I stacked a few 28pin dip sockets and added a zif socket in my book8088 so I can easily swap out the BIOS chip and have two chips I can switch between for testing. I am happy to test any builds you want. I got this so I can test the FreeDOS kernel more easily.

Here's a ROM image build I made that disabled the KB errors and includes the XTIDE and GLaTICK for RTC ISA:
https://github.com/640-KB/GLaBIOS/releases/download/v0.2.5/GLABIOS_0.2.6_VB_XTIDE-GLaTICK_AT.ROM

Is there anything else you would add to this?  If you want to try this one out, let me know if this would be a better candidate image to send to anyone else testing it out?

Thanks!

**Comment by PerditionC** [comment]
That works great, thank you.  No need to press F1 for keyboard errors,
clock & time set correctly on boot, and from running a few basic tests,
everything seems to work as well as with the included BIOS (better since
this build works with my RTC).

**Comment by 640-KB** [comment]
> That works great, thank you. No need to press F1 for keyboard errors, clock & time set correctly on boot, and from running a few basic tests, everything seems to work as well as with the included BIOS (better since this build works with my RTC).

Fantastic news -- thank you for trying that out.  If you run into anything odd please let me know!

640KB

**Comment by AWalGH** [comment]
Recently got one of these funny Book PCs and have been tinkering with various things. After jumping through a few hoops I've been able to build the BIOS in MASM6, and I have a working build chain that throws a BIOS ROM together that includes XUB, GlaTICK and the IBM CBASIC if I'm feeling frisky.

I've noticed that if there's an FPU installed it will usually work anyways despite the boot screen still stating none. I only have one Intel D8087-1 to check with at the moment, and it's breaking VGA init unless it's been on for a little bit, so I think there's something wrong with it...Unfortunately the only other systems I could test an 8087 with are in storage currently.

More importantly, I can't seem to get it to boot at all on an AMD D8088. I _think_ it's boot-looping, but nothing is showing on screen other than the LCDs firmware stating no video.

I'm using a v2 model if that even matters. I have the CGA card on order that should be here in a few days, though I expect that might not actually change much...

**Comment by AWalGH** [comment]
Did a bit of further tinkering and my builds were not interpreting the settings I was trying to pass in the command line. Installing MASM 5.0 and using that instead of 6 fixed that and suddenly booting using the AMD8088 and accompanying ROM build is working fine with it now.

The FPU still seems to mess with the VGAs init, and I'm starting to suspect this might be timing related as it only seems to fudge the video mode and font, but enough power cycles brings it back...I suppose this could be related to the VGA BIOS as well...As previously the issue doesn't happen with an FPU installed, and doesn't appear to happen at all with the provided BIOS.

> Interesting! Do you think you could share the ROM image you are using? I have a V1 and I can test it out on mine.

Sure. The ROM stack is XUB - GLaTICK - Empty Space - CBASIC - GLaBIOS

I don't have an RTC but the line shows up like it should when GLaBIOS inits and XUB's F8 ROMBOOT option does go into CBASIC

[https://krakenfiles.com/view/iDXmEUes0I/file.html](GLaBIOS_BOOK8088_8B_240318.7z)

> Out of curiosity, have you tried the ROMs in the [releases](https://github.com/640-KB/GLaBIOS/releases/v0.2.5#beta) for Book8088, just to see if it runs at all on yours?

Having a chance to try the pre-build ROMs I can confirm the same behavior with the V20 build.
The 8088 build appears to just work, so I suspect my trouble was with MASM6 this whole time.
